### PR TITLE
Add Key/Value JSON Builder support

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR adds the JSON Builder which allows for users to create simple, top-level JSON objects for use with CDOGS. This builder does not support nested structures.
- Add official google material icon support
- Create editable table field component
- Improve reactive behavior for buttons with small screens
- Add tabbed view for contexts generation

This change allows less technical users to be able to create a simple JSON that can be used with CDOGS.

[[SHOWCASE-374]](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-374)

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wfezkf-tools.pathfinder.gov.bc.ca/dashboard?id=document-generation-showcase-pr-20

File: app/frontend/src/components/FileInput.vue [hunk 1]
```diff
@@ -1,35 +1,64 @@
...
-          auto-grow
-          hint="JSON format for key-value pairs"
-          label="Contexts"
-          :mandatory="true"
-          required
-          :rules="contextsRules"
-          v-model="contexts"
-        />
+        <v-row>
+          <v-col cols="12" md="4">
```
Comment (loneil) [inline app/frontend/src/components/FileInput.vue]: Do you need the cols="12", or will it default to that below the md breakpoint? I can't remember.

File: app/frontend/src/components/JsonBuilder.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          <v-btn @click="deleteItem(item)" fab class="mb-2" small text v-on="on">
+            <v-icon>delete</v-icon>
+          </v-btn>
+        </template>
+        <span>Add Key/Value Pair</span>
+      </v-tooltip>
+    </template>
+    <template v-slot:footer>
+      <v-container>
+        <v-row justify="end">
```
Comment (loneil) [inline app/frontend/src/components/JsonBuilder.vue]: Does this need to be in a row? No responsive columns.

File: app/frontend/src/components/JsonBuilder.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+<script>
+export default {
+  name: 'jsonbuilder',
+  data() {
+    return {
+      headers: [
+        { text: 'Key', value: 'key' },
+        { text: 'Value', value: 'value' },
+        { text: 'Action', value: 'action' }
+      ],
```
Comment (loneil) [inline app/frontend/src/components/JsonBuilder.vue]: Unused now that the table header is gone?

File: app/frontend/src/App.vue [hunk 4]
```diff
@@ -29,8 +29,8 @@ export default {
 
 <style lang="scss" scoped>
 body {
-  margin: 0;
-  padding: 0;
+  // margin: 0;
```
[the remark this answers is not in the document]
Comment (jujaga) [reply app/frontend/src/App.vue]: Probably not. I left them in as I don't know if they're supposed to affect something else I'm not aware of. I'll remove them.

File: app/frontend/src/components/FileInput.vue (hunk 1 above)
Comment (jujaga) [reply app/frontend/src/components/FileInput.vue]: If cols="12" isn't there, it won't collapse the columns correctly on smaller screen sizes - it doesn't do that by default.

File: app/frontend/src/components/JsonBuilder.vue (hunk 2 above)
Comment (jujaga) [reply app/frontend/src/components/JsonBuilder.vue]: This was one of the only succinct ways I could pull the button to the right side of the pane without resorting to some ugly CSS hacks.

File: app/frontend/src/components/JsonBuilder.vue (hunk 3 above)
Comment (jujaga) [reply app/frontend/src/components/JsonBuilder.vue]: This is still used internally in the table even if it isn't shown on the page. I tried removing it and the entire table broke horribly.

**Pull request merged**